packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -0,0 +1,241 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualNode } from './virtual-node';
+
+/**
+ * Properties for the Weighted Targets in the route
+ */
+export interface WeightedTarget {
+  /**
+   * The VirtualNode the route points to
+   */
+  readonly virtualNode: IVirtualNode;
+
+  /**
+   * The weight for the target
+   *
+   * @default 1
+   */
+  readonly weight?: number;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface HttpRouteMatch {
+  /**
+   * Specifies the path to match requests with.
+   * This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   */
+  readonly prefixPath: string;
+}
+
+/**
+ * Properties specific for HTTP Based Routes
+ */
+export interface HttpRouteSpecOptions {
+  /**
+   * The criterion for determining a request match for this Route
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpRouteMatch;
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+
+  /**
+   * The priority for the route. Routes are matched based on the specified value, where 0 is the highest priority.
+   *
+   * @default - none
+   */
+  readonly priority?: number;
+}
+
+/**
+ * Properties specific for a TCP Based Routes
+ */
+export interface TcpRouteSpecOptions {
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+}
+
+/**
+ * All Properties for GatewayRoute Specs
+ */
+export interface RouteSpecConfig {
+  /**
+   * The spec for an http route
+   *
+   * @default - no http spec
+   */
+  readonly httpRouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for an http2 route
+   *
+   * @default - no http2 spec
+   */
+  readonly http2RouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for a grpc route
+   *
+   * @default - no grpc spec
+   */
+  readonly grpcRouteSpec?: CfnRoute.GrpcRouteProperty;
```

Comment:
Could easily implement this in this PR, but I figure it may be better in a separate PR

Reply:
Sorry, what is "this" here? 🙂

Reply:
GRPC Route Specs

Reply:
I'm fine with adding them in this PR 🙂